### ISSUE TITLE
Running demo.py with other videos

### ISSUE THREAD

**Issue by Einhartd** [bharath5673/YOLOv8-3D#7]
When I try to change sample video and run demo.py 2D boxes work quite well in detecting cars but 3D doesn't work at all.
I've run this video on mobilenetv2.
![image](https://github.com/bharath5673/YOLOv8-3D/assets/87729500/1714fda6-3601-4fbf-beac-5fb1e56a5cad)
These are the primary parameters of this new video:
![image](https://github.com/bharath5673/YOLOv8-3D/assets/87729500/4169b460-07ee-402e-9857-ac2e13ef9092)
I have also tested this on vgg19 with the same result.
How should I add new demo video to this model?
Is this problem related to camera parameters? If so, where can I change them?

**Comment by j-coching** [comment]
hello, i like to ask how you were able to run your demo.py file? I have my weights file. but, i'm getting a JIT compilation failed error. Thanks!

**Comment by ChAstral0** [comment]
This is the error I'm getting when I run:

File "/home/erdt/Desktop/YOLOv8-3D/demo.py", line 227, in <module>

  File "/home/erdt/Desktop/YOLOv8-3D/demo.py", line 178, in process3D

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 65, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 2655, in predict

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 2440, in predict_function

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 2425, in step_function

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 2413, in run_step

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 2381, in predict_step

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 65, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/training.py", line 590, in __call__

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 65, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/base_layer.py", line 1149, in __call__

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 96, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/functional.py", line 515, in call

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/functional.py", line 672, in _run_internal_graph

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 65, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/engine/base_layer.py", line 1149, in __call__

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 96, in error_handler

  File "/home/erdt/anaconda3/envs/test1/lib/python3.10/site-packages/keras/src/layers/core/lambda_layer.py", line 212, in call

  File "/home/erdt/Desktop/YOLOv8-3D/train.py", line 277, in l2_normalize

JIT compilation failed.
         [[{{node model/orientation/l2_normalize/Rsqrt}}]] [Op:__inference_predict_function_10189]

**Comment by Einhartd** [comment]
I haven't encountered this error so I have no idea how to help you with this problem.

**Comment by adnan10tc39** [comment]
can i fine-tune this on my custom dataset?

**Comment by bharath5673** [comment]
> When I try to change sample video and run demo.py 2D boxes work quite well in detecting cars but 3D doesn't work at all. I've run this video on mobilenetv2. ![image](https://private-user-images.githubusercontent.com/87729500/288559902-1714fda6-3601-4fbf-beac-5fb1e56a5cad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjE0ODAsIm5iZiI6MTcwMzA2MTE4MCwicGF0aCI6Ii84NzcyOTUwMC8yODg1NTk5MDItMTcxNGZkYTYtMzYwMS00ZmJmLWJlYWMtNWZiMWU1NmE1Y2FkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4MzMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1ZTQ2NzJkOWM2NmM0MzhkOTYzYzE5MWEzYjliOTI1NzExYjUwMWFlMGZlNmYyM2E1YTUxOTRmNThiZWQzZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xlXFoOlZPpn-mFXQsweoxMR36wbBfNP9uX92BZtIn5w) These are the primary parameters of this new video: ![image](https://private-user-images.githubusercontent.com/87729500/288560212-4169b460-07ee-402e-9857-ac2e13ef9092.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjE0ODAsIm5iZiI6MTcwMzA2MTE4MCwicGF0aCI6Ii84NzcyOTUwMC8yODg1NjAyMTItNDE2OWI0NjAtMDdlZS00MDJlLTk4NTctYWMyZTEzZWY5MDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4MzMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZDQ0ZGI4ZTgxYTIwNWQ2MTdjZjNiY2JjOTY0MDBhMzdhN2QwZmY1NzNmNDRhODQ3YTM1ZTlhYzYxOWE2MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M4rhXdLaWLsq-VVz8i0wcobiYyi_Co8IwSXOKX6CuQw) I have also tested this on vgg19 with the same result. How should I add new demo video to this model? Is this problem related to camera parameters? If so, where can I change them?

@Einhartd .. yes that is because wrong/diff camera calibration.. Camera calibration is like teaching ur camera how to see things.. and involves determining the intrinsic parameters (internal characteristics) and extrinsic parameters (external positioning) of the camera, and it plays a crucial role in transforming 2D image coordinates into real-world 3D coordinates especially while camera moves or rotates... usually cameras can introduce distortions, such as radial and tangential distortions, which affect the accuracy of measurements which mismatches this kitti camera matrix..

**Comment by bharath5673** [comment]
> > When I try to change sample video and run demo.py 2D boxes work quite well in detecting cars but 3D doesn't work at all. I've run this video on mobilenetv2. ![image](https://private-user-images.githubusercontent.com/87729500/288559902-1714fda6-3601-4fbf-beac-5fb1e56a5cad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjE0ODAsIm5iZiI6MTcwMzA2MTE4MCwicGF0aCI6Ii84NzcyOTUwMC8yODg1NTk5MDItMTcxNGZkYTYtMzYwMS00ZmJmLWJlYWMtNWZiMWU1NmE1Y2FkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4MzMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1ZTQ2NzJkOWM2NmM0MzhkOTYzYzE5MWEzYjliOTI1NzExYjUwMWFlMGZlNmYyM2E1YTUxOTRmNThiZWQzZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xlXFoOlZPpn-mFXQsweoxMR36wbBfNP9uX92BZtIn5w) These are the primary parameters of this new video: ![image](https://private-user-images.githubusercontent.com/87729500/288560212-4169b460-07ee-402e-9857-ac2e13ef9092.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjE0ODAsIm5iZiI6MTcwMzA2MTE4MCwicGF0aCI6Ii84NzcyOTUwMC8yODg1NjAyMTItNDE2OWI0NjAtMDdlZS00MDJlLTk4NTctYWMyZTEzZWY5MDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4MzMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZDQ0ZGI4ZTgxYTIwNWQ2MTdjZjNiY2JjOTY0MDBhMzdhN2QwZmY1NzNmNDRhODQ3YTM1ZTlhYzYxOWE2MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M4rhXdLaWLsq-VVz8i0wcobiYyi_Co8IwSXOKX6CuQw) I have also tested this on vgg19 with the same result. How should I add new demo video to this model? Is this problem related to camera parameters? If so, where can I change them?
> 
> @Einhartd .. yes that is because wrong/diff camera calibration.. Camera calibration is like teaching ur camera how to see things.. and involves determining the intrinsic parameters (internal characteristics) and **extrinsic parameters** (external positioning) of the camera, and it plays a crucial role in transforming 2D image coordinates into real-world 3D coordinates especially while camera moves or rotates... usually cameras can introduce distortions, such as radial and tangential distortions, which affect the accuracy of measurements which mismatches this kitti camera matrix..




_Here are the general steps to add a new demo video to a model and address potential camera parameter issues:_

**Capture or Obtain New Video:**
Record or obtain the new video that you want to use for the demo.

**Check Camera Parameters:**
Verify the camera parameters used during training. These parameters typically include the intrinsic matrix (focal length, principal point) and distortion coefficients.

**Calibrate Camera:**
If the new video is captured with a different camera, you may need to calibrate the camera to obtain its intrinsic parameters. Follow the camera calibration steps.
 
**Update Model Configuration:**
Once you have the new camera parameters, update the model configuration

**Comment by Einhartd** [comment]
@bharath5673 Thank you so much for your response. Can I use kitti raw dataset as a video with calibration parameters? If so, where can I update/change this camera parameter in code?

**Comment by bharath5673** [comment]
> @bharath5673 Thank you so much for your response. Can I use kitti raw dataset as a video with calibration parameters? If so, where can I update/change this camera parameter in code?

[here](https://github.com/bharath5673/YOLOv8-3D/blob/219c9ae639135e384ee672238e2a47f090bd93e4/demo.py#L35)

**Comment by HassaanIrfan1** [comment]
> > [@bharath5673](https://github.com/bharath5673) Thank you so much for your response. Can I use kitti raw dataset as a video with calibration parameters? If so, where can I update/change this camera parameter in code?
> 
> [here](https://github.com/bharath5673/YOLOv8-3D/blob/219c9ae639135e384ee672238e2a47f090bd93e4/demo.py#L35)

Hey, @bharath5673. I hope you will be going well.
I replaced this P2 matrix with my camera calibration matrix that I found from ROS camera calibration for monocular camera. It still shows disoriented 3d bounding boxes. Can you help me regarding this?
